### PR TITLE
Add Fx Option Trade and Fx Option Product

### PR DESCRIPTION
**Pull request by santorog**
The notion of FxOptionTrade as a marker interface would be useful, hence this merge request.

### CODE REVIEW

**Review by jodastephen — CHANGES_REQUESTED** [review]
There also needs to be an `FxOptionProduct` interface that extends `FxProduct` and adds a `getExpiry() method. `FxOptionProduct` would then override `getProduct()` to return `FxOptionProduct`

File: modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxSingleBarrierOption.java [hunk 1]
```diff
@@ -115,14 +116,22 @@ private void validate() {
...
     return underlyingOption.getCurrencyPair();
   }
 
+  //-----------------------------------------------------------------------
+  /**
+   * Gets the underlying Fx vanilla option's expiry.
+   * @return the expiry
+   */
+  @Override
+  public ZonedDateTime getExpiry() { return underlyingOption.getExpiry(); }
```
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxSingleBarrierOption.java:133]: This formatting is wrong.

File: modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxSingleBarrierOption.java [hunk 2]
```diff
@@ -59,15 +60,15 @@
...
+   * The barrier level stored in this field must be represented based on the direction of the currency pair in the
    * underlying FX transaction.
    * <p>
    * For example, if the underlying option is an option on EUR/GBP, the barrier should be a certain level of EUR/GBP rate.
    */
   @PropertyDefinition(validate = "notNull")
   private final Barrier barrier;
   /**
-   * For a 'out' option, the amount is paid when the barrier is reached; 
+   * For a 'out' option, the amount is paid when the barrier is reached;
```
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxSingleBarrierOption.java:71]: Your IDE should not be configured to automatically remove whitespace like this. (No need to undo the whitespace parts)

File: modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxVanillaOption.java [hunk 3]
```diff
@@ -49,7 +49,7 @@
  */
 @BeanDefinition
 public final class FxVanillaOption
-    implements FxProduct, Resolvable<ResolvedFxVanillaOption>, ImmutableBean, Serializable {
+    implements FxOptionProduct, Resolvable<ResolvedFxVanillaOption>, ImmutableBean, Serializable {
```
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxVanillaOption.java:52]: Need to add `@Override` to `getExpiry()`

**Review by jodastephen — CHANGES_REQUESTED** [review]

File: modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxSingleBarrierOption.java (hunk 2 above)
Comment (santorog) [reply modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxSingleBarrierOption.java:71]: I followed the dev guide to configured my IDE and just double-checked I did that correctly earlier, I am not sure what is happening.

**Review by jodastephen — APPROVED** [review]

**Pull request merged**